app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/GoogleCloudBuildStageForm.tsx

```diff
@@ -60,17 +68,60 @@ export class GoogleCloudBuildStageForm extends React.Component<
     }));
   };
 
+  private getCloudBuildTriggers = (): Array<Option<string>> => {
+    return this.props.gcbTriggers.map(trigger => ({
+      label: trigger.name,
+      value: trigger.id,
+    }));
+  };
+
+  private getTriggerTypes = (): Array<Option<string>> => {
+    return [
+      { value: 'branchName', label: 'Branch name' },
+      { value: 'tagName', label: 'Tag Name' },
+      { value: 'commitSha', label: 'Commit SHA' },
+    ];
+  };
+
   private getSourceOptions = (): Array<Option<string>> => {
     return [
       { value: buildDefinitionSources.TEXT, label: 'Text' },
       { value: buildDefinitionSources.ARTIFACT, label: 'Artifact' },
+      { value: buildDefinitionSources.TRIGGER, label: 'Trigger' },
     ];
   };
 
   private onSourceChange = (e: any): void => {
     this.props.formik.setFieldValue('buildDefinitionSource', e.target.value);
   };
 
+  private onTriggerTypeValueChange = (e: any): void => {
+    const stage = this.props.formik.values;
+    if (stage.triggerType) {
+      const path = `repoSource.${stage.triggerType}`;
+      this.props.formik.setFieldValue(path, e.target.value);
+    }
+  };
+
+  private onTriggerChange = (selectedTrigger: Option) => {
+    this.props.formik.setFieldValue('triggerId', selectedTrigger.value);
+  };
+
+  private onTriggerTypeChange = (selectedType: Option) => {
+    this.props.formik.setFieldValue('triggerType', selectedType.value);
+    this.props.formik.setFieldValue('repoSource', undefined);
+  };
+
+  private getTriggerTypeValue = () => {
+    const stage = this.props.formik.values;
+    if (stage.triggerType) {
+      const path = `repoSource.${stage.triggerType}`;
+      return get(stage, path);
+    }
+
+    return undefined;
```

Comment:
```suggestion
    return '';
```

Since this is setting the input value, let's default to an empty string.